lib/active_merchant/billing/gateways/ogone.rb

```diff
@@ -460,7 +460,7 @@ def calculate_signature(signed_parameters, algorithm, secret)
             raise "Unknown signature algorithm #{algorithm}"
           end
 
-        filtered_params = signed_parameters.compact
+        filtered_params = signed_parameters.reject { |_k, v| v.nil? || v == '' }
```

Comment:
I recommend `v.blank?` because it is true when the value is `nil?` or it is an empty string

Reply:
I thought the same but it doesn't work with booleans:

![Screenshot 2023-11-20 at 8 43 32 PM](https://github.com/activemerchant/active_merchant/assets/133580/5ecae9b7-81a9-4e17-89d1-d1fdfb23b3b0)